File: src/main/java/vitaliy94/bookshoptest/model/Book.java
package vitaliy94.bookshoptest.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Positive;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Book
{
    int id;
    @NotBlank
    String name;
    @Positive
    int year;
    @NotBlank
    String description;
    @Positive
    double price;
}

File: front/src/app/components/serializable/book.ts
export class Book {
  constructor(public id: number,
              public name: string,
              public year: number,
              public description: string,
              public price: number) {}

  // public clone() {
  //   return new Book(this.id, this.name, this.year, this.description, this.price);
  // }
}
